Tools/ardupilotwaf/git_submodule.py

```diff
@@ -111,8 +112,28 @@ def u(s):
     def __str__(self):
         return 'Submodule update: %s' % self.submodule
 
+def check_folder(path_list=[]):
+    for d in Utils.to_list(path_list):
+        listdir = os.listdir(d)
+        for f in listdir(d):
+            if f.startsith('.'):
+                listdir.remove()
+        if len(f) == 0:
```

Comment:
If this is to remove `.` and `..` then explicitly check them them.

Reply:
This is done in order to remove certain hidden files that might appear in an otherwise empty folder. (For example: .DS_store and .git)

Reply:
Do we *want* to touch a folder which has *any* entries except `.` and `..`?!